model/task_history.go

```diff
@@ -779,12 +781,60 @@ func testHistoryV2Results(params *TestHistoryParameters) ([]task.Task, error) {
 	tasks = append(tasks, oldTasks...)
 
 	// to join the test results, merge test results for all the tasks
-	testQuery := db.Query(formTestsQuery(params))
-	out, err := task.MergeTestResultsBulk(tasks, &testQuery)
-	if err != nil {
-		return nil, errors.Wrap(err, "error merging test results")
+	if len(params.TestNames) > 0 {
+		testQuery := db.Query(formTestsQuery(params))
+		out, err := task.MergeTestResultsBulk(tasks, &testQuery)
+		if err != nil {
+			return nil, errors.Wrap(err, "error merging test results")
+		}
+		return out, nil
+	} else {
```

Comment:
I would either: 

- take out this else statement, or
- use var statements to declare out and err above line 783, and then have one return statement after the else closes

Reply:
Took out the else part